src/main/java/org/minions/devfund/noemiguzman/battleship/Ship.java

```diff
@@ -0,0 +1,202 @@
+     * @param column number
+     * @param horizontal true
+     * @param ocean object
+     */
+    public void placeShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        bowRow = row;
+        bowColumn = column;
+        this.horizontal = horizontal;
+
+        /*
+         * Set references to the ship in the ships array in the Ocean object
+         */
+        if (horizontal) {
+            for (int j = column; j < column + length; j++) {
+                ocean.getShipArray()[row][j] = this;
+            }
+        } else {
+            for (int i = row; i < row + length; i++) {
+                ocean.getShipArray()[i][column] = this;
+            }
+        }
+    }
+
+
+    /**
+     * Return true if the part of the ship was hit, false otherwise.
+     * @param row number
+     * @param column number
+     * @return true
+     */
+    public boolean wasShootAt(int row, int column) {
+        return horizontal ? hit[column - this.bowColumn] : hit[row - this.bowRow];
+    }
+
+    /**
+     * If a part of the ship occupies the given row and column, and the ship hasnot been sunk,
+     * mark that part of the ship as hit (in the hit array, 0 indicates the bow) and return true,
+     * otherwise return false.
+     * @param row number
+     * @param column number
+     * @return true
+     */
+    public boolean shootAt(int row, int column) {
+        if (isSunk()) {
+            return false;
+        }
+        final boolean isHittable = horizontal ? row == this.bowRow && column < this.bowColumn + length
+                : column == this.bowColumn && row < this.bowRow + length;
+        if (isHittable) {
+            final int hitPosition = horizontal ? column - this.bowColumn : row - this.bowRow;
+            hit[hitPosition] = true;
+            return true;
+        }
+        return false;
+    }
+
+
+    /**
+     * Return true if every part of the ship has been hit, false otherwise.
+     * @return true
+     */
+    public boolean isSunk() {
+        for (boolean b : hit) {
+            if (!b) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        return isSunk() ? "x" : "S";
+    }
+
+
+    /**
+     * set size.
+     * @param length int
+     */
+    public void setLength(int length) {
+        this.length = length;
+    }
+
+    /**
+     * returns hit.
+     * @return true hit
+     */
+    public boolean[] getHit() {
+        return hit.clone();
+    }
+
+    /**
+     * set hit.
+     * @param hit object
+     */
+    public void setHit(final boolean[] hit) {
+        this.hit = hit.clone();
+    }
+
+    /**
+     * put ship on ocean.
+     * @param ocean object
+     */
+    public void putRandom(final Ocean ocean) {
+        Random random = new Random();
+        int row;
+        int column;
+        boolean horizontalRando;
+        while (true) {
+            row = random.nextInt(ocean.getSizeOcean());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this instance-reference to a static reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)